demo_nodes_py/demo_nodes_py/events/matched_event_detect.py

```diff
@@ -17,91 +17,185 @@
-        if self.connect_subscription:
+    def __pub_matched_event_callback(self, info: QoSPublisherMatchedInfo):
+        if self.__any_subscription_connected:
             if info.current_count == 0:
                 self.get_logger().info('Last subscription is disconnected.')
-                self.connect_subscription = False
+                self.__any_subscription_connected = False
             else:
-                self.get_logger().info('Current number of connected subscription is '
+                self.get_logger().info('The changed number of connected subscription is '
+                                       + str(info.current_count_change) + ' and '
+                                       'current number of connected subscription is '
                                        + str(info.current_count))
         else:
             if info.current_count != 0:
                 self.get_logger().info('First subscription is connected.')
-                self.connect_subscription = True
+                self.__any_subscription_connected = True
 
-    def sub_matched_event_callback(self, info: QoSSubscriptionMatchedInfo):
-        if self.connect_subscription:
+    def __sub_matched_event_callback(self, info: QoSSubscriptionMatchedInfo):
+        if self.__any_publisher_connected:
             if info.current_count == 0:
                 self.get_logger().info('Last publisher is disconnected.')
-                self.connect_subscription = False
+                self.__any_publisher_connected = False
             else:
-                self.get_logger().info('Current number of connected publisher is '
+                self.get_logger().info('The changed number of connected publisher is '
+                                       + str(info.current_count_change) + ' and current '
+                                       'number of connected publisher is '
                                        + str(info.current_count))
         else:
             if info.current_count != 0:
                 self.get_logger().info('First publisher is connected.')
-                self.connect_subscription = True
+                self.__any_publisher_connected = True
 
-    def sub_callback(self, msg):
+    def __sub_callback(self, msg):
         True
 
 
+class MultiSubNode(Node):
+
+    def __init__(self, topic_name: String):
+        super().__init__('multi_sub_node')
+        self.__subs = []
+        self.__topic_name = topic_name
+
+    def create_one_sub(self) -> Subscription:
+        self.get_logger().info('Create a new subscription.')
+        sub = self.create_subscription(String, self.__topic_name, self.__sub_callback, 10)
+        self.__subs.append(sub)
+        return sub
+
+    def destroy_one_sub(self, sub: Subscription):
+
+        if sub in self.__subs:
+            self.get_logger().info('Destroy a subscription.')
+            self.__subs.remove(sub)
+            self.destroy_subscription(sub)
+
+    def __sub_callback(self, msg):
+        True
```

Comment:
Generally, I think we will use `pass` instead of `True` here.
a better way to use an empty callback for the subscription might use
```
sub = self.create_subscription(String, self.__topic_name, lambda msg: ..., 10)
```
and then remove `__sub_callback`.

Reply:
After the above update, I'll trigger the CI.

Reply:
Yes. Revised it.